[CI] ClassificationIT testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword failing

**Build Scans:**
- [elasticsearch-periodic #6068 / openjdk23_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/uw2y3crav7eva)
- [elasticsearch-pull-request #53649 / part-4](https://gradle-enterprise.elastic.co/s/jtjjgflxlya54)
- [elasticsearch-pull-request #52861 / part-4](https://gradle-enterprise.elastic.co/s/tcrx66xjwt7mm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.ClassificationIT.testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword" -Dtests.seed=DFFCAD7EBFDCC68B -Dtests.locale=pis-Latn-SB -Dtests.timezone=Pacific/Ponape -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ClassificationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Stats were: {"id":"classification_training_percent_is_50_keyword","state":"started","progress":[{"phase":"reindexing","progress_percent":100},{"phase":"loading_data","progress_percent":100},{"phase":"feature_selection","progress_percent":100},{"phase":"coarse_parameter_search","progress_percent":84},{"phase":"fine_tuning_parameters","progress_percent":0},{"phase":"final_training","progress_percent":0},{"phase":"writing_results","progress_percent":0},{"phase":"inference","progress_percent":0}],"data_counts":{"training_docs_count":150,"test_docs_count":0,"skipped_docs_count":0},"memory_usage":{"timestamp":1738157265776,"peak_usage_bytes":19080,"status":"ok"},"node":{"id":"DC5ygAvfTtm6RgbgLCL2rQ","name":"javaRestTest-1","ephemeral_id":"QTExINPyQvmsXQNG-DNkmw","transport_address":"127.0.0.1:35695","attributes":{"ml.machine_memory":"126604201984","testattr":"test","xpack.installed":"true","transform.config_version":"10.0.0","ml.allocated_processors":"32","ml.alloca
[truncated]
```

**Issue Reasons:**
- [main] 3 failures in test testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword (0.6% fail rate in 521 executions)
- [main] 2 failures in step part-4 (0.8% fail rate in 258 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.8% fail rate in 258 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -254,9 +254,6 @@ tests:
 - class: org.elasticsearch.xpack.security.FileSettingsRoleMappingsRestartIT
   method: testFileSettingsReprocessedOnRestartWithoutVersionChange
   issue: https://github.com/elastic/elasticsearch/issues/120964
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/120071
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testGetUsersWithProfileUidWhenProfileIndexDoesNotExists
   issue: https://github.com/elastic/elasticsearch/issues/121179
@@ -265,9 +262,6 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSetEnabled
   issue: https://github.com/elastic/elasticsearch/issues/121183
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithDatastreams
-  issue: https://github.com/elastic/elasticsearch/issues/121236
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=transform/*}
   issue: https://github.com/elastic/elasticsearch/issues/120816
@@ -317,9 +311,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/121151
 - class: org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/121407
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDependentVariableIsAliasToNested
-  issue: https://github.com/elastic/elasticsearch/issues/121415
 - class: org.elasticsearch.xpack.security.authc.jwt.JwtRealmSingleNodeTests
   method: testClientSecretRotation
   issue: https://github.com/elastic/elasticsearch/issues/120985
@@ -347,9 +338,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only}
   issue: https://github.com/elastic/elasticsearch/issues/121412
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDependentVariableIsAliasToKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/121492
 - class: org.elasticsearch.search.CrossClusterSearchUnavailableClusterIT
   method: testSearchSkipUnavailable
   issue: https://github.com/elastic/elasticsearch/issues/121497
@@ -372,9 +360,6 @@ tests:
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   method: test {yaml=snapshot.delete/10_basic/Delete a snapshot asynchronously}
   issue: https://github.com/elastic/elasticsearch/issues/122102
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsBoolean
-  issue: https://github.com/elastic/elasticsearch/issues/121680
 - class: org.elasticsearch.xpack.downsample.DownsampleActionSingleNodeTests
   method: testDuplicateDownsampleRequest
   issue: https://github.com/elastic/elasticsearch/issues/122158

